README.md

```diff
@@ -20,25 +20,29 @@ requires:
     interface: bind-client
 ```
 
-## flag: {relation_name}.available
+## flag: {relation_name}.connected
 
 This flag is set when the subordinate unit is joined and removes when departed.
 
 For example, the following code will update the relation just when the relation is estabilished and the configuration changed:
 
 ```python
-if reactive.is_flag_set('bind-stats.available'):
+@reactive.when('rndckey.available')
+@reactive.when('dns-backend.related')
+def config_changed():
+    '''Render configs and restart services if necessary'''
+    endpoints = [
+        reactive.endpoint_from_flag('dns-backend.related'),
+        reactive.endpoint_from_name('bind-stats')
+    ]
```

Comment:
This code needs to specify that for the example, the `metadata.yaml` file of this charm would have a "dns-backend" binding and a "bind-stats" endpoint using the `bind-client` interface. Otherwise, it is not clear where the "dns-backend.related" flag or the "bind-stats" endpoint come from. It is also unclear where "designate_bind" object comes from (lines 44-45).

OTOH, I would split this list in separate variables (see example below). This is because calling endpoints[1] is not immediately understandable (line 42).
```
dns_backend_endpoint = reactive.endpoint_from_flag("dns-backend.related")
bind_stats_endpoint = reactive.endpoint_from_name("bind-stats")
config = hookenv.config()

if reactive_is_flag_set("bind-stats.connected") and config.changed("stats-port"):
    bind_stats_endpoint.configure(config["stats-port"])

# etc.
```